### PR TITLE
feat: support property have defined getter or setter, @InputBoolean s…

### PR DESCRIPTION
**Pull request by LinBoLen**
…till take effect

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by wilsoncook** [comment]
Hi @LinBoLen , thanks for your pr.

I wonder the use-cases for this pr that why we need to use the original getter/setter, could you provide some use cases to support this change, thank you!

File: components/core/util/convert.ts [hunk 1]
```diff
@@ -31,27 +31,35 @@ export function valueFunctionProp<T>(prop: FunctionProp<T>, ...args: any[]): T {
...
+          return this[privatePropName]; // tslint:disable-line:no-invalid-this
+        },
+        set(value: boolean | string): void {
+          this[privatePropName] = toBoolean(value); // tslint:disable-line:no-invalid-this
+        }
+      });
+    } else {
+      Object.defineProperty(target, name, {
+        get(): boolean {
+          return originalOwnPropertyDescriptor.get.call(target); // tslint:disable-line:no-invalid-this
```
Comment (wilsoncook) [inline components/core/util/convert.ts]: The original `set/get` can also be `undefined`

Comment (LinBoLen) [reply components/core/util/convert.ts]: this is just a wrapper for have defined originalOwnPropertyDescriptor, if The original set/get can also be undefined, then call origin undefined getter setter. 
it works as expect

**Comment by vthinkxie** [comment]
Due to no progress for a long time, this issue will be closed. Please contact us if you have any questions.

**Pull request closed without merging**